### ISSUE TITLE
[Bug]: Quests Editor annoyances

### ISSUE THREAD

**Issue by Abalieno** [FTBTeam/FTB-Mods-Issues#829]
### Mod

FTB Quests

### Mod version

ftb-quests-fabric-1902.4.15-build.228

### Forge / Fabric version

Fabric 0.14.19

### What issue are you having?

Not tested with latest version, but I didn't see anything relevant in the logs.

There are a couple problems:

- While typing in the editor, if the line exceeds the right margin of the window, all text disappears. Exiting and reentering shows everything again, but it has to be redone every time this happens. Usually only once for each line. It only happens when the current line first exceeds the margin.

- While typing, if you then go select a color tag with the mouse, the text editor loses focus. If you then press the letter "e", the editor closes and deletes all the text you typed up to that point. I suppose because after clicking on a color it interprets it as the command menu having focus, and then thinks "e" is exit or something. So it closes without saving. Problem is, when I type and select a color I then expect to be able to type again, because my mind is focused on what I'm writing. The fact you have to select a color and then remember you have to click back on the line you were typing to give it back focus is annoying. In standard text editor if you're typing, then select the font to be bold, you can continue typing immediately. You don't lose focus. But the fact that if what you type contains the letter "e", auto-closes the editor and discards everything you wrote up to that point is definitely a major issue.

### Steps to reproduce

Within the description.

**Comment by desht** [comment]
Problem 1 has actually also been reported by someone else, but so far I've been completely unable to reproduce it. No matter what I've tried, typing beyond the end of the line always inserts a soft break for me and continues on the next line, as expected.  I'll continue to try to reproduce it, but any more info you have about the circumstances in which it happens would be appreciated.

Problem 2 is one I'll get fixed for the next build.  I agree, it's a poor UX right now.

**Comment by Abalieno** [comment]
I reproduced the other problem of the window going "blank".

It doesn't happen at the end of a line when you continue typing, because that jumps on the next line normally. The problem is if you press "space". I can easily reproduce by just getting at the border of the window and then adding spaces.

**Comment by desht** [comment]
I still can't reproduce this. Even at the window edge and inserting spaces, it still behaves as expected...

**Comment by Abalieno** [comment]
Then it's probably related to the game window size. I'm currently used a window of 1360x832

If I go fullscreen and then back the problem clears out, so it's probably some logic depending how the border of the quest editor is handled.

**Comment by desht** [comment]
Yeah... except I've also tested with a wide variety of window sizes :(

I will keep this open, and maybe one day I'll be able to reproduce it.  But I just can't make it break right now.
